### DISCUSSION TITLE
Add on-chain calculation for bond repay amount needed for liquidations

### DISCUSSION THREAD

**Discussion by scorpion9979** [hifi-finance/hifi#42] [Ideas]
Currently, the way [`liquidateBorrow(...)`](https://github.com/hifi-finance/hifi/blob/0eeff4c32b51d9f4ba6e26b369e04561297dbf4b/packages/protocol/contracts/core/balanceSheet/IBalanceSheetV1.sol#L192) works makes it much easier to liquidate a multi-bond single collateral vault than a single bond multi-collateral vault. This is due to the fact that when a vault is to be liquidated, the client app would need to construct a nested for loop where one loop iterates over all borrowed bond tokens from that vault and the other loop iterates over all the collateralized assets in that vault, the order of loop nesting being negligible. There exists an issue with liquidations that didn't exist with the previous iteration of the protocol that only dealt with single collaterals. The issue is met when the full borrowed bond amount would yield more collateral to be liquidated than the actual collateral, therefore the client would not be able to use the value returned by [`getDebtAmount(...)`](https://github.com/hifi-finance/hifi/blob/0eeff4c32b51d9f4ba6e26b369e04561297dbf4b/packages/protocol/contracts/core/balanceSheet/IBalanceSheetV1.sol#L102) as the `repayAmount` param in `liquidateBorrow(...)`, because the `repayAmount` needed to liquidate the entire `collateral` asset in the vault would be less than the borrowed bond amount in the case of multi-collateral. To resolve this, I suggest we add a basic on-chain getter method for calculating the `repayAmount` needed to be repaid in order to liquidate a specified amount of a certain `collateral` as follows:

```solidity
function getRepayBondAmount(
        IHToken bond,
        uint256 liquidationAmount,
        IErc20 collateral
    ) public view override returns (uint256 repayBondAmount) {
   ...
}
```

**Comment by PaulRBerg** [comment]
This feature has been added in https://github.com/hifi-finance/hifi/pull/38

Locking this convo.
